pandas/core/frame.py

```diff
@@ -6616,16 +6621,53 @@ def _gotitem(
         **_shared_doc_kwargs
     )
     @Appender(_shared_docs["aggregate"])
-    def aggregate(self, func, axis=0, *args, **kwargs):
+    def aggregate(self, func=None, axis=0, *args, **kwargs):
         axis = self._get_axis_number(axis)
 
+        relabeling = func is None and _is_multi_agg_with_relabel(**kwargs)
+        if relabeling:
+            func, indexes, order = _normalize_keyword_aggregation(kwargs)
+            reordered_indexes = [
+                pair[0] for pair in sorted(zip(indexes, order), key=lambda t: t[1])
+            ]
```

Comment:
most of this is already implemented in the SelectionMixin
you should not have to add much code

Reply:
emm, i think this implementation is sort of post-processing after aggregation from SelectionMixin? no? 
i will try my best to shorten the code anyway! thanks! @jreback